bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/ItemGroup.java

```diff
@@ -38,7 +39,7 @@ public class ItemGroup {
      * Keep the list of items that should run on same thread.
      * Using a stack to get advantage of the locality.
      */
-    Stack<SchedulerItem> items = new Stack<>();
+    Deque<SchedulerItem> items = new ConcurrentLinkedDeque<>();
```

Comment:
This will be only access through same thread. Hence do we need ConcurrentLinkedDeque?

Reply:
May be ArrayDeque?